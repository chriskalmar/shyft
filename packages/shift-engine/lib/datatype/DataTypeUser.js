
import DataType from './DataType';


export default class DataTypeUser extends DataType {}

export const isDataTypeUser = (obj) => {
  return (obj instanceof DataTypeUser)
}
