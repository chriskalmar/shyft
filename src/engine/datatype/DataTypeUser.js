import DataType from './DataType';

export class DataTypeUser extends DataType {}

export const isDataTypeUser = obj => {
  return obj instanceof DataTypeUser;
};
