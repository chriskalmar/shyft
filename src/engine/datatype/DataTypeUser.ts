import { DataType } from './DataType';

export class DataTypeUser extends DataType {}

export const isDataTypeUser = (obj: unknown): obj is DataTypeUser => {
  return obj instanceof DataTypeUser;
};
