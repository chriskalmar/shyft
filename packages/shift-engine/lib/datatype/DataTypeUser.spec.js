
import { assert } from 'chai';
import DataTypeUser, { isDataTypeUser } from './DataTypeUser';

import {
  passOrThrow,
} from '../util';


describe('DataTypeUser', () => {

  describe('isDataTypeUser', () => {


    it('should recognize objects of type DataTypeUser', () => {

      const dataTypeUser = new DataTypeUser({
        name: 'user',
        description: 'some description',
        mock() {},
      })

      function fn() {
        passOrThrow(
          isDataTypeUser(dataTypeUser),
          () => 'This error will never happen'
        )
      }

      assert.doesNotThrow(fn)

    })


    it('should recognize non-DataTypeUser objects', () => {

      function fn() {
        passOrThrow(
          isDataTypeUser({}) ||
          isDataTypeUser(function test() {}) ||
          isDataTypeUser(assert),
          () => 'Not a DataType object'
        )
      }


      assert.throws(fn, /Not a DataType object/);

    })

  })


})
