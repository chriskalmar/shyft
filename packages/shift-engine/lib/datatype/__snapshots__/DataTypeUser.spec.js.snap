// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTypeUser isDataTypeUser should recognize non-DataTypeUser objects 1`] = `"Not a DataType object"`;
